flow/layers/container_layer.cc

```diff
@@ -129,6 +129,9 @@ void ContainerLayer::PrerollChildren(PrerollContext* context,
   FML_DCHECK(!context->has_platform_view);
   bool child_has_platform_view = false;
   bool child_has_texture_layer = false;
+  bool subtree_can_accept_opacity =
+      container_can_pass_opacity_to_children() && layers_.size() < 2;
```

Comment:
Should we have a TODO to further optimize this e.g. with RTrees?

Reply:
Bounds tracking and culling is already an optimization mentioned in other issues.

Reply:
Filed https://github.com/flutter/flutter/issues/93899

Reply:
I realized that I could optimize the case where we have a linear list of children laid out next to each other with very little code, which is actually quite a number of cases for many apps that tend to have lists of things and other row/column style layouts. A more sophisticated (i.e. RTree) algorithm could go deeper, but this should catch a number of cases as long as the layout is simple and every child can accept the opacity hand-me-down.